pkg/nfd-master/nfd-master.go

```diff
@@ -553,49 +552,38 @@ func (m *nfdMaster) filterFeatureLabels(labels Labels, features *nfdv1alpha1.Fea
 }
 
 func (m *nfdMaster) filterFeatureLabel(name, value string, features *nfdv1alpha1.Features) (string, error) {
-	//Validate label name
-	if errs := k8svalidation.IsQualifiedName(name); len(errs) > 0 {
-		return "", fmt.Errorf("invalid name %q: %s", name, strings.Join(errs, "; "))
+	// Check if Value is dynamic
+	var filteredValue string
+	if strings.HasPrefix(value, "@") {
+		dynamicValue, err := getDynamicValue(value, features)
+		if err != nil {
+			return "", err
+		}
+		filteredValue = dynamicValue
+	} else {
+		filteredValue = value
 	}
 
-	// Check label namespace, filter out if ns is not whitelisted
+	// Validate
 	ns, base := splitNs(name)
-	if ns == "" {
-		return "", fmt.Errorf("labels without namespace (prefix/) not allowed")
-	}
-	if ns != nfdv1alpha1.FeatureLabelNs && ns != nfdv1alpha1.ProfileLabelNs &&
-		!strings.HasSuffix(ns, nfdv1alpha1.FeatureLabelSubNsSuffix) && !strings.HasSuffix(ns, nfdv1alpha1.ProfileLabelSubNsSuffix) {
+	err := validate.Label(name, filteredValue)
+	if err == validate.ErrNSNotAllowed {
```

Comment:
With the change I suggested for validate.go I'd modify this one:

```suggestion
	if err == validate.ErrNSNotAllowed || isNamespaceDenied(ns, m.deniedNs.wildcard, m.deniedNs.normal) {
```

This would change the behavior slightly so that it would become possible to deny the default `feature.node.kubernetes.io` and `profile.node.kubernetes.io` but I don't believe that would be a breaking anyone. Just document it in `docs/reference/master-configuration-reference.md` (basically remove the note that the default namespaces cannot be denied). WDYT?